pkg/reconciler/knativeeventing/controller.go

```diff
@@ -35,14 +36,17 @@ import (
 )
 
 const (
-	controllerAgentName = "knativeeventing-controller"
-	reconcilerName      = "KnativeEventing"
+	controllerAgentName       = "knativeeventing-controller"
+	reconcilerName            = "KnativeEventing"
+	ChannelBasedBrokerClass   = "ChannelBasedBroker"
+	MTChannelBasedBrokerClass = "MTChannelBasedBroker"
 )
 
 var (
-	recursive  = flag.Bool("recursive", false, "If filename is a directory, process all manifests recursively")
-	MasterURL  = flag.String("master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
-	Kubeconfig = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+	recursive             = flag.Bool("recursive", false, "If filename is a directory, process all manifests recursively")
+	MasterURL             = flag.String("master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	Kubeconfig            = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+	AvailableBrokerClases = []string{ChannelBasedBrokerClass, MTChannelBasedBrokerClass}
```

Comment:
not sure if available is "right" here.  

There might be others "available" ? I think I still could install my "custom broker", and set its class on the CR, right ? 

Perhaps built-in? :man_shrugging:

Reply:
hmm, never thought of that.
good point. gonna change some things to support it